src/signatures.jl

```diff
@@ -53,9 +53,10 @@ function signature(@nospecialize(recurse), frame::Frame, @nospecialize(stmt), pc
         stmt = pc_expr(frame, pc)
     end
     isa(stmt, Expr) || return nothing, pc
+    mt = extract_method_table(frame, stmt; eval = false)
     sigsv = @lookup(frame, stmt.args[2])::SimpleVector
     sigt = signature(sigsv)
-    return sigt, lastpc
+    return mt => sigt, lastpc
```

Comment:
```suggestion
    return Pair{Union{Nothing,MethodTable},Any}(mt, sigt), lastpc
```
Since the type of `sigt` is dynamic, we should try to avoid arbitrary runtime type of `Pair` objects (and it might be a better idea to restrict the type of `signatures` to `signatures::Vector{Pair{Union{Nothing,MethodTable}}` accordingly).